### PR TITLE
Add to channel dialog + story + container

### PR DESCRIPTION
**Pull request by buoyad**
This is ready for review. r? @keybase/react-hackers

### CODE REVIEW

**Comment by buoyad** [comment]
@keybase/react-hackers Added the call to the RPC and waiting/error states. Doing rendering of the system message in a separate PR

File: shared/chat/conversation/info-panel/add-to-channel/container.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
+  const users = stateProps._allMembers
+    .map(username => ({
+      alreadyAdded: stateProps._alreadyAdded.includes(username),
+      fullname: stateProps._fullnames.get(username, {fullname: ''}).fullname,
+      username,
+    }))
+    .sort((a, b) => {
+      if (a.alreadyAdded && b.alreadyAdded) return a.username.localeCompare(b.username)
```
Comment (akalin-keybase) [inline shared/chat/conversation/info-panel/add-to-channel/container.js]: can do

```
if (a.alreadyAdded == b.alreadyAdded) return ...
return a.alreadyAdded ? 1 : -1
```

Comment (akalin-keybase) [inline shared/chat/conversation/info-panel/add-to-channel/container.js]: can do

```
if (a.alreadyAdded == b.alreadyAdded) return ...
return a.alreadyAdded ? 1 : -1
```

File: shared/chat/conversation/info-panel/add-to-channel/index.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+              </Kb.Text>
+            )}
+          </Kb.Box2>
+        </Kb.Box2>
+      }
+      action={
+        <Kb.Checkbox
+          disabled={user.alreadyAdded}
+          checked={user.alreadyAdded || user.selected}
+          onCheck={() => this._toggleSelected(user.username)}
```
Comment (akalin-keybase) [inline shared/chat/conversation/info-panel/add-to-channel/index.js]: clicking the checkbox doesn't do anything but clicking the row does -- i suspect  `_toggleSelected` is getting triggered twice

File: shared/common-adapters/list2.native.js [hunk 3]
```diff
@@ -50,6 +50,7 @@ class List2<T> extends PureComponent<Props<T>, void> {
           onEndReached={this.props.onEndReached}
           windowSize={this.props.windowSize || 10}
           debug={false /* set to true to debug the list */}
+          contentContainerStyle={this.props.style}
```
Comment (akalin-keybase) [inline shared/common-adapters/list2.native.js]: will this affect any other existing styles?

File: shared/chat/conversation/info-panel/add-to-channel/index.js (hunk 2 above)
Comment (akalin-keybase) [reply shared/chat/conversation/info-panel/add-to-channel/index.js]: only on desktop

File: protocol/bin/enabled-calls.json [hunk 4]
```diff
@@ -1,4 +1,5 @@
 {
+  "chat.1.local.bulkAddToConv": {"promise": true},
```
Comment (akalin-keybase) [inline protocol/bin/enabled-calls.json]: nothing seems to happen when I click 'Add X user[s]'. Is the core side supposed to be plumbed through already?

File: shared/chat/conversation/info-panel/add-to-channel/container.js (hunk 1 above)
Comment (buoyad) [reply shared/chat/conversation/info-panel/add-to-channel/container.js]: nice :+1:

File: shared/chat/conversation/info-panel/add-to-channel/index.js (hunk 2 above)
Comment (buoyad) [reply shared/chat/conversation/info-panel/add-to-channel/index.js]: unfortunate that this happens on desktop, `checkbox.desktop` should probably have a `stopPropagation()` there. I'll just switch this handler off on desktop.

File: shared/common-adapters/list2.native.js (hunk 3 above)
Comment (buoyad) [reply shared/common-adapters/list2.native.js]: No, nothing else uses this prop

File: protocol/bin/enabled-calls.json (hunk 4 above)
Comment (buoyad) [reply protocol/bin/enabled-calls.json]: Yeah it should be plumbed through. I see the button spin and then on success the thread we were adding users to.

**Pull request merged**